gum/backend-windows/gumtls-windows.c

```diff
@@ -28,6 +53,166 @@ gum_tls_key_free (GumTlsKey key)
   TlsFree (key);
 }
 
+
+void
+_gum_tls_init (void)
+{
+#if defined (HAVE_I386)
+  gum_spinlock_init (&_gum_tls_tmp_keys_lock);
+  memset (_gum_tls_tmp_keys, 0, sizeof (_gum_tls_tmp_keys));
+#endif
+}
+
+#if defined (HAVE_I386)
+
+static gpointer
+_gum_tls_key_get_tmp_value (GumTlsKey key)
+{
+  guint i;
+  GumThreadId tid = gum_process_get_current_thread_id ();
+  gpointer value = NULL;
+
+  gum_spinlock_acquire (&_gum_tls_tmp_keys_lock);
+
+  for (i = 0; i < MAX_TMP_TLS_KEY; i++)
```

Comment:
The pattern we're using for for-loops is always `i != MAX_TMP_TLS_KEY` as it's easier to detect off-by-one errors (it will result in busy-loop instead of weird bugs).